src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -641,7 +642,7 @@ private static void SendEHelloCallback(IAsyncResult result)
                             if ((e.StatusCode != SmtpStatusCode.CommandUnrecognized)
                                 && (e.StatusCode != SmtpStatusCode.CommandNotImplemented))
                             {
-                                throw e;
+                                throw;
```

Comment:
`throw` retains the information but not `throw e`? Why don't we need the Capture/Throw here?